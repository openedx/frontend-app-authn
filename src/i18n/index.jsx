import { messages as paragonMessages } from '@edx/paragon';

import arMessages from './messages/ar.json';
import deMessages from './messages/de.json';
import dedeCAMessages from './messages/de_DE.json';
import es419Messages from './messages/es_419.json';
import frMessages from './messages/fr.json';
import frCAMessages from './messages/fr_CA.json';
import hiMessages from './messages/hi.json';
import itMessages from './messages/it.json';
import ititCAMessages from './messages/it_IT.json';
import ptMessages from './messages/pt.json';
import ptptCAMessages from './messages/pt_PT.json';
import ruMessages from './messages/ru.json';
import ukMessages from './messages/uk.json';
import zhcnMessages from './messages/zh_CN.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  'es-419': es419Messages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': dedeCAMessages,
  'it-it': ititCAMessages,
  'pt-pt': ptptCAMessages,
};

export default [
  paragonMessages,
  appMessages,
];
