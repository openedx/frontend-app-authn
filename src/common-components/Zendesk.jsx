import React from 'react';

import { getConfig } from '@edx/frontend-platform';
import { useIntl } from '@edx/frontend-platform/i18n';
import Zendesk from 'react-zendesk';

import messages from './messages';

const ZendeskHelp = () => {
  const { formatMessage } = useIntl();
  const setting = {
    cookies: true,
    webWidget: {
      contactOptions: {
        enabled: false,
      },
      chat: {
        suppress: false,
      },
      contactForm: {
        ticketForms: [
          {
            id: 360003368814,
            subject: false,
            fields: [{ id: 'description', prefill: { '*': '' } }],
          },
        ],
        selectTicketForm: {
          '*': formatMessage(messages.selectTicketForm),
        },
        attachments: true,
      },
      helpCenter: {
        originalArticleButton: true,
      },
      answerBot: {
        suppress: false,
        contactOnlyAfterQuery: true,
        title: { '*': formatMessage(messages.supportTitle) },
        avatar: {
          url: getConfig().ZENDESK_LOGO_URL,
          name: { '*': formatMessage(messages.supportTitle) },
        },
      },
    },
  };

  return (
    <Zendesk defer zendeskKey={getConfig().ZENDESK_KEY} {...setting} />
  );
};

export default ZendeskHelp;
