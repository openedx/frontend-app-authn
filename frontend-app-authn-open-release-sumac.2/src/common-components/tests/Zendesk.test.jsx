import { IntlProvider } from '@edx/frontend-platform/i18n';
import renderer from 'react-test-renderer';

import Zendesk from '../Zendesk';

jest.mock('react-zendesk', () => 'Zendesk');

describe('Zendesk Help', () => {
  it('should match login page third party auth alert message snapshot', () => {
    const tree = renderer.create(
      <IntlProvider locale="en">
        <Zendesk />
      </IntlProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
