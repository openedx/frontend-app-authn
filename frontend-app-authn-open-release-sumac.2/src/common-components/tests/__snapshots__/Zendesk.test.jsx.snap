// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zendesk Help should match login page third party auth alert message snapshot 1`] = `
<Zendesk
  cookies={true}
  defer={true}
  webWidget={
    {
      "answerBot": {
        "avatar": {
          "name": {
            "*": "edX Support",
          },
          "url": undefined,
        },
        "contactOnlyAfterQuery": true,
        "suppress": false,
        "title": {
          "*": "edX Support",
        },
      },
      "chat": {
        "departments": {
          "enabled": [
            "account settings",
            "billing and payments",
            "certificates",
            "deadlines",
            "errors and technical issues",
            "other",
            "proctoring",
          ],
        },
        "suppress": false,
      },
      "contactForm": {
        "attachments": true,
        "selectTicketForm": {
          "*": "Please choose your request type:",
        },
        "ticketForms": [
          {
            "fields": [
              {
                "id": "description",
                "prefill": {
                  "*": "",
                },
              },
            ],
            "id": 360003368814,
            "subject": false,
          },
        ],
      },
      "contactOptions": {
        "enabled": false,
      },
      "helpCenter": {
        "originalArticleButton": true,
      },
    }
  }
/>
`;
